File: Models/PromizzApp.Models/SearchBaseModel.cs
using System.Collections.Generic;

namespace PromizzApp.Models
{
    public class SearchBaseModel
    {
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public int TotalElements { get; set; }
        public int TotalPages { get; set; }
        public string OrderBy { get; set; }
        public string OrderDir { get; set; }
        public string SearchText { get; set; }
    }

    public class SearchResult<T> where T : class
    {
        public IEnumerable<T> data;
    }
}

File: PromizzApp.Client/src/app/shared/models/searchbase.model.ts
export class SearchBaseModel {
    public PageNumber: number = 0;
    public PageSize: number = 15;
    public TotalElements: number = 0;
    public TotalPages: number = 0;
    public OrderBy: string = '';
    public OrderDir: string = 'asc';
    public SearchText: string = '';

    public constructor(init?: Partial<SearchBaseModel>) {
        Object.assign(this, init);
    }
}

export class SearchResult<T> {
    public data: Array<T>;
}
